discord/interactions.py

```diff
@@ -1419,3 +1449,145 @@ def __init__(self, *, data: MessageInteractionPayload, state: ConnectionState):
+        The metadata of the interaction that opened the model. Only present on interactions of type
+        :attr:`InteractionType.modal_submit`.
+    """
+
+    __slots__: tuple[str, ...] = (
+        "id",
+        "type",
+        "user",
+        "authorizing_integration_owners",
+        "original_response_message_id",
+        "interacted_message_id",
+        "triggering_interaction_metadata",
+        "_state",
+        "_cs_original_response_message",
+        "_cs_interacted_message",
+    )
+
+    def __init__(self, *, data: InteractionMetadataPayload, state: ConnectionState):
+        self._state = state
+        self.id: int = int(data["id"])
+        self.type: InteractionType = try_enum(InteractionType, data["type"])
+        self.user: User = User(state=state, data=data["user"])
+        self.authorizing_integration_owners: AuthorizingIntegrationOwners = (
+            AuthorizingIntegrationOwners(data["authorizing_integration_owners"], state)
+        )
+        self.original_response_message_id: int | None = utils._get_as_snowflake(
+            data, "original_response_message_id"
+        )
+        self.interacted_message_id: int | None = utils._get_as_snowflake(
+            data, "interacted_message_id"
+        )
+        self.triggering_interaction_metadata: InteractionMetadata | None = None
+        if tim := data.get("triggering_interaction_metadata"):
+            self.triggering_interaction_metadata = InteractionMetadata(
+                data=tim, state=state
+            )
+
+    def __repr__(self):
+        return (
+            f"<InteractionMetadata id={self.id} type={self.type!r} user={self.user!r}>"
+        )
+
+    @utils.cached_slot_property("_cs_original_response_message")
+    def original_response_message(self) -> Message | None:
+        """Optional[:class:`Message`]: The original response message.
+        Returns ``None`` if the message is not in cache, or if :attr:`original_response_message_id` is ``None``.
+        """
+        if not self.original_response_message_id:
+            return None
+        return self._state._get_message(self.original_response_message_id)
+
+    @utils.cached_slot_property("_cs_interacted_message")
+    def interacted_message(self) -> Message | None:
+        """Optional[:class:`Message`]: The message that triggered the interaction.
+        Returns ``None`` if the message is not in cache, or if :attr:`interacted_message_id` is ``None``.
+        """
+        if not self.interacted_message_id:
+            return None
+        return self._state._get_message(self.interacted_message_id)
+
+
+class AuthorizingIntegrationOwners:
+    """Contains details on the authorizing user or server for the installation(s) relevant to the interaction.
+
+    Attributes
+    ----------
+    user_id: :class:`int` | :class:`MISSING`
+        The ID of the user that authorized the integration.
+    guild_id: :class:`int` | None | :class:`MISSING`
```

Comment:
Why are we using MISSING here?

Reply:
`guild_id` is nullable and also optional, so this reflects that.
It was used for `user_id` as well for consistency

Reply:
We don't usually use `MISSING` on front-facing attributes though, it only represents edit fields that are to remain untouched.